// jest-express/lib/router.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const request_1 = require("./request");
const response_1 = require("./response");
class Router {
    constructor() {
        this.request = new request_1.Request();
        this.response = new response_1.Response();
        this.all = jest.fn();
        this.get = jest.fn((path, callback) => {
            if (typeof path === 'string' && typeof callback === 'function') {
                callback(this.request, this.response);
            }
            else {
                path(this.request, this.response);
            }
        });
        this.param = jest.fn();
        this.route = jest.fn(() => {
            return {
                get: this.get,
            };
        });
        this.use = jest.fn();
        return this;
    }
    resetMocked() {
        this.all.mockReset();
        this.get.mockReset();
        this.param.mockReset();
        this.route.mockReset();
        this.use.mockReset();
    }
}
exports.Router = Router;
